src/geometry/rectangle.rs

```diff
@@ -0,0 +1,186 @@
+use nalgebra as na;
+use super::traits::{Transform, Plane};
+use super::super::config::*;
+use super::super::error::*;
+use super::utils;
+
+/// A struct for sensors of rectangular geometry
+#[derive(Debug)]
+pub struct Rectangle {
+    pub gloabl_center: P3,  //center of the sensor (not used in bound checks)
+    pub normal : Vec3,      // normal vector of plane
+    pub plane_constant: Real, // D in Ax +By + Cz +D =0 
+
+    half_base: Real,
+    half_height: Real,
+
+    to_global: Aff3,    // L => G for point
+    to_local: Aff3,     // G => L for point
+    measurement_to_state_vector: Mat2x5
+}
+
+impl Rectangle {
+    /*
+    /// This is the constructor for the rectangular geometry. It expects a 4x4 `nalgebra::Matrix4<f32>` that is invertible 
+    /// and a 4 element array of `nalgebra::Point3<f32>`. If the matrix is not invertible it will return `Err(&str)`.
+    /// The provided matrix should be an affine transformation for converting from R2->R3
```

Comment:
Nitpick: the affine transform transforms from R3 -> R3, it's just that we then choose to ignore the z coordinate, thus effectively turning it into R2.

Reply:
updated comment